src/tools/unstable-book-gen/src/main.rs

```diff
@@ -11,13 +13,121 @@ use tidy::unstable_book::{
     LIB_FEATURES_DIR, PATH_STR,
 };
 
-fn generate_stub_issue(path: &Path, name: &str, issue: u32) {
-    let content = format!(include_str!("stub-issue.md"), name = name, issue = issue);
+use rustdoc_json_types::{Crate, Id};
+use syn::{parse::Parser, Ident, Lit, Meta, NestedMeta};
+
+// (item path, doc url)
+fn full_path(krate: &Crate, item: &Id) -> Option<(String, String)> {
+    let item_summary = krate.paths.get(item)?;
+    let kind = &item_summary.kind;
+    let kind_str = serde_json::to_string(kind).ok()?;
+    let mut url = String::from("https://doc.rust-lang.org/nightly/");
+    let mut iter = item_summary.path.iter();
+    iter.next_back();
+    url.push_str(&iter.cloned().collect::<Vec<_>>().join("/"));
+    url.push('/');
+    url.push_str(kind_str.trim_matches('"'));
+    url.push('.');
+    url.push_str(item_summary.path.last().unwrap());
+    url.push_str(".html");
+    Some((item_summary.path.join("::"), url))
+}
```

Comment:
Is it only me that this looks hard to debug/understand the workflow inside? :)

Maybe we can have some comments to inform developer about what's going on in there. + Using `expect` instead of `unwrap` can be nice to have as well.

Reply:
This tries (quite badly) to generate the URL that the item belongs to.
If there is a way to get it directly from Rustdoc, I will happily use it.
Concerning the unwraps, this is mostly prototype code.

Reply:
> This tries (quite badly) to generate the URL that the item belongs to.
If there is a way to get it directly from Rustdoc, I will happily use it.

Having comments which explains aim of the function instructions(which are complicated to read/track) can be handy imo.(e.g https://github.com/rust-lang/rust/blob/91eb6f9acfcfde6832d547959ad2d95b1ac0b5dc/src/bootstrap/builder.rs#L1165-L1930)

> Concerning the unwraps, this is mostly prototyping code that is not meant to last.

It's just to give more clear information in context. Not so necessary I guess.